src/logic/map_objects/tribes/tribes.cc

```diff
@@ -35,145 +35,30 @@
-	i18n::Textdomain td("tribes");
-	buildings_->add(new MarketDescr(
-	   pgettext_expr(table.get_string("msgctxt").c_str(), table.get_string("descname").c_str()),
-	   table, *this));
-}
-
-void Tribes::add_immovable_type(const LuaTable& table) {
-	i18n::Textdomain td("tribes");
-	immovables_->add(new ImmovableDescr(
-	   pgettext_expr(table.get_string("msgctxt").c_str(), table.get_string("descname").c_str()),
-	   table, *this));
-}
-
-void Tribes::add_ship_type(const LuaTable& table) {
-	i18n::Textdomain td("tribes");
-	ships_->add(new ShipDescr(_(table.get_string("descname")), table));
-}
-
-void Tribes::add_ware_type(const LuaTable& table) {
-	i18n::Textdomain td("tribes");
-	wares_->add(new WareDescr(
-	   pgettext_expr(table.get_string("msgctxt").c_str(), table.get_string("descname").c_str()),
-	   table));
-}
-
-void Tribes::add_carrier_type(const LuaTable& table) {
-	i18n::Textdomain td("tribes");
-	workers_->add(new CarrierDescr(
-	   pgettext_expr(table.get_string("msgctxt").c_str(), table.get_string("descname").c_str()),
-	   table, *this));
-}
-
-void Tribes::add_ferry_type(const LuaTable& table) {
-	i18n::Textdomain td("tribes");
-	workers_->add(new FerryDescr(
-	   pgettext_expr(table.get_string("msgctxt").c_str(), table.get_string("descname").c_str()),
-	   table, *this));
-}
-
-void Tribes::add_soldier_type(const LuaTable& table) {
-	i18n::Textdomain td("tribes");
-	workers_->add(new SoldierDescr(
-	   pgettext_expr(table.get_string("msgctxt").c_str(), table.get_string("descname").c_str()),
-	   table, *this));
-}
-
-void Tribes::add_worker_type(const LuaTable& table) {
-	i18n::Textdomain td("tribes");
-	workers_->add(new WorkerDescr(
-	   pgettext_expr(table.get_string("msgctxt").c_str(), table.get_string("descname").c_str()),
-	   table, *this));
-}
-
-void Tribes::add_tribe(const LuaTable& table, const World& world) {
-	const std::string name = table.get_string("name");
-	if (Widelands::tribe_exists(name)) {
-		tribes_->add(new TribeDescr(table, Widelands::get_tribeinfo(name), world, *this));
-	} else {
-		throw GameDataError("The tribe '%s'' has no preload file.", name.c_str());
-	}
-}
-
-void Tribes::add_custom_building(const LuaTable& table) {
-	const std::string tribename = table.get_string("tribename");
-	if (Widelands::tribe_exists(tribename)) {
-		TribeDescr* descr = tribes_->get_mutable(tribe_index(tribename));
-		const std::string buildingname = table.get_string("buildingname");
-		descr->add_building(buildingname);
-	} else {
-		throw GameDataError("The tribe '%s'' has no preload file.", tribename.c_str());
+     largest_workarea_(0),
+     scenario_tribes_(nullptr),
+     lua_(lua),
+     description_manager_(new Widelands::DescriptionManager(lua_)) {
+
+	// Register tribe names. Tribes have no attributes.
+	std::vector<std::string> attributes;
+	for (const TribeBasicInfo& tribeinfo : Widelands::get_all_tribeinfos()) {
+		description_manager_->register_description(tribeinfo.name, tribeinfo.script, attributes);
 	}
```

Comment:
`assert(attributes.empty());`

Reply:
We can't assert anything about contents of Lua files. So, I'll throw a GameDataError instead.